requests/__init__.py

```diff
@@ -86,6 +87,25 @@ def check_compatibility(urllib3_version, chardet_version):
 except ImportError:
     pass
 
+# Check for older version of pyOpenSSL
+def check_pyopenssl(cryptography_version, pyopenssl_version):
+    # Only applies to pyOpenSSL
+    if sys.modules['OpenSSL'].__title__ == 'pyOpenSSL':
+        # Warning only applies for Cryptography < 1.3.4
+        if list(map(int, cryptography_version.split('.'))) < [1, 3, 4]:
+            # Warning only applies for pyOpenSSL < 16.0.0
+            assert list(map(int, cryptography_version.split('.'))) < [16, 0, 0]
+
+# Check cryptography version
+if 'OpenSSL' in sys.modules:
+    try:
+        cryptography_version = sys.modules['cryptography'].__version__
+        pyopenssl_version = sys.modules['OpenSSL'].__version__
+        check_pyopenssl(cryptography_version, pyopenssl_version)
+    except (AssertionError, ValueError):
+        warning = 'Old version of pyOpenSSL ({0}) may cause slowdown.'.format(pyopenssl_version)
```

Comment:
Is the issue for old versions of `pyOpenSSL` or `cryptography`.

Reply:
Old versions of `pyOpenSSL`. @nateprewitt recommended checking `Cyptography`'s version before `pyOpenSSL` in [this comment](https://github.com/requests/requests/issues/4309#issuecomment-332942003).

Reply:
To clarify, I believe the comment says we want to verify `pyOpenSSL` is installed, and if it is check that `cryptography` is at least 1.3.4 which is our minimum requirement. We haven't done any testing to see how far back you need to go with `cryptography` for the slowdown to occur, but 1.3.4 should be sufficient.